dartagnan/src/main/antlr4/LitmusC.g4

```diff
@@ -95,90 +99,94 @@ re locals [IOpBin op, String mo]
         | AtomicFetchDec        LPar address = re RPar {$op = IOpBin.SUB; $mo = Linux.MO_MB;}
         | AtomicFetchDecRelaxed LPar address = re RPar {$op = IOpBin.SUB; $mo = Linux.MO_RELAXED;}
         | AtomicFetchDecAcquire LPar address = re RPar {$op = IOpBin.SUB; $mo = Linux.MO_ACQUIRE;}
-        | AtomicFetchDecRelease LPar address = re RPar {$op = IOpBin.SUB; $mo = Linux.MO_RELEASE;})                   # reAtomicFetchOp
+        | AtomicFetchDecRelease LPar address = re RPar {$op = IOpBin.SUB; $mo = Linux.MO_RELEASE;})                                     # reAtomicFetchOp
 
     |   ( AtomicXchg        LPar address = re Comma value = re RPar {$mo = Linux.MO_MB;}
         | AtomicXchgRelaxed LPar address = re Comma value = re RPar {$mo = Linux.MO_RELAXED;}
         | AtomicXchgAcquire LPar address = re Comma value = re RPar {$mo = Linux.MO_ACQUIRE;}
-        | AtomicXchgRelease LPar address = re Comma value = re RPar {$mo = Linux.MO_RELEASE;})                          # reXchg
+        | AtomicXchgRelease LPar address = re Comma value = re RPar {$mo = Linux.MO_RELEASE;})                                          # reXchg
 
     |   ( Xchg        LPar address = re Comma value = re RPar {$mo = Linux.MO_MB;}
         | XchgRelaxed LPar address = re Comma value = re RPar {$mo = Linux.MO_RELAXED;}
         | XchgAcquire LPar address = re Comma value = re RPar {$mo = Linux.MO_ACQUIRE;}
-        | XchgRelease LPar address = re Comma value = re RPar {$mo = Linux.MO_RELEASE;})                                # reXchg
+        | XchgRelease LPar address = re Comma value = re RPar {$mo = Linux.MO_RELEASE;})                                                # reXchg
 
     |   ( AtomicCmpXchg        LPar address = re Comma cmp = re Comma value = re RPar {$mo = Linux.MO_MB;}
         | AtomicCmpXchgRelaxed LPar address = re Comma cmp = re Comma value = re RPar {$mo = Linux.MO_RELAXED;}
         | AtomicCmpXchgAcquire LPar address = re Comma cmp = re Comma value = re RPar {$mo = Linux.MO_ACQUIRE;}
-        | AtomicCmpXchgRelease LPar address = re Comma cmp = re Comma value = re RPar {$mo = Linux.MO_RELEASE;})        # reCmpXchg
+        | AtomicCmpXchgRelease LPar address = re Comma cmp = re Comma value = re RPar {$mo = Linux.MO_RELEASE;})                        # reCmpXchg
 
-    |   C11AtomicSCAS LPar address = re Comma expectedAdd = re Comma value = re Comma c11Mo Comma c11Mo RPar            # reC11SCmpXchg
-    |   C11AtomicWCAS LPar address = re Comma expectedAdd = re Comma value = re Comma c11Mo Comma c11Mo RPar            # reC11WCmpXchg
+    |   C11AtomicSCASExplicit LPar address = re Comma expectedAdd = re Comma value = re Comma c11Mo Comma c11Mo openCLScope? RPar       # reC11SCmpXchgExplicit
+    |   C11AtomicSCAS         LPar address = re Comma expectedAdd = re Comma value = re RPar                                            # reC11SCmpXchg
```

Comment:
why not optional scope? 
Same for rule below

Reply:
Based on https://man.opencl.org/atomic_compare_exchange.html it seems scope is only expected when Mo are explicitly set.

Reply:
👍 ... I was not aware of that

Reply:
> EDIT: it seems you are missing scopes in pretty much all atomics

Thanks! added.